src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -324,7 +331,15 @@ function VisualizeAppController(
   // appState then they won't be equal.
   if (!_.isEqual(stateContainer.getState().vis, stateDefaults.vis)) {
     try {
-      vis.setState(stateContainer.getState().vis);
+      const visState = stateContainer.getState().vis;
+      const state = {
+        type: visState.type,
+        params: visState.params,
+        data: {
+          aggs: visState.aggs,
+        },
+      };
+      vis.setState(state);
```

Comment:
If you only need to extract `aggs` and put them into `data`, you could simplify this:
```
const { vis: { aggs, ...visState } }= stateContainer.getState();
vis.setState({ ...visState, data: { aggs } });

Reply:
And you also missed a `title` here, which was presented in `savedVisState` object:

![image](https://user-images.githubusercontent.com/31325372/76947651-fe9b2980-6916-11ea-84e1-6b48ef3f8731.png)